internal/handler/msg_aggregate.go

```diff
@@ -42,6 +42,10 @@ import (
 
 // MsgAggregate implements `aggregate` command.
 func (h *Handler) MsgAggregate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
+	if err := h.authenticate(ctx); err != nil {
```

Comment:
One alternative to adding this code wherever authentication is desired is to make authentication opt-out rather than opt-in. You might want to consider this to avoid needing to add this everywhere as this is a common security fragility (eventually someone might forget to add this on a new endpoint).

So, for endpoints that require no authentication, we could require them to be wrapped by a function on commands.go. For example, say MsgPing doesn't require authentication. Rather than

```go
"ping": {
	Handler: h.MsgPing,
	Help:    "Returns a pong response.",
},
```

We could have to write it as the following for it to work for unauthenticated users:

```go
"ping": {
	Handler: UnsafeHandler(h.MsgPing),
	Help:    "Returns a pong response.",
},
```

or

```go
"ping": {
	Handler: h.MsgPing,
	Unsafe: true,
	Help:    "Returns a pong response.",
},

// With a Unsafe boolean, it'd be easy to check programmatically what commands
// doesn't require authentication.
// Maybe useful to write a test to automatically check it & maintain a list of
// allowed commands for security purposes.
```

And then UnsafeHandler would mark that the handler as safe to call without authentication.

By the way, am I correct to assume most handlers need authentication, btw? :)